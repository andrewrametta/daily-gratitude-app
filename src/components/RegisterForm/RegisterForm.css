.password-directions {
  color: rgb(0 0 0 / 50%);
}
