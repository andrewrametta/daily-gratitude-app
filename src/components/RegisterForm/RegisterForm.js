import React, { Component } from "react";
import AuthAPIService from "../../services/auth-api-service";
import "./RegisterForm.css";

class Register extends Component {
  state = {
    error: null,
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { username, password, confirmPassword } = e.target;
    this.setState({ error: null });
    if (password.value === confirmPassword.value) {
      AuthAPIService.postUser({
        username: username.value,
        password: password.value,
      })
        .then((user) => {
          this.props.history.push("/login");
        })
        .catch((res) => {
          this.setState({ error: res.error });
        });
    } else {
      alert("Passwords do not match");
    }
  };

  render() {
    return (
      <div>
        <section>
          <h3>Register Now</h3>

          <form className="register-form" onSubmit={this.handleSubmit}>
            {this.state.error && <p className="error">{this.state.error}</p>}
            <div>
              <label htmlFor="username">Username </label>
              <input type="text" name="username" id="username" />
            </div>
            <div>
              <label htmlFor="password">Password </label>
              <input type="password" name="password" id="password" />
            </div>
            <div>
              <label htmlFor="confirm-password">Confirm Password </label>
              <input
                type="password"
                name="confirmPassword"
                id="confirm-password"
              />
            </div>
            <p className="password-directions">
              Password must be at least 8 characters and include a number, a
              capital letter, and a special character
            </p>
            <button type="submit">Register</button>
          </form>
        </section>
      </div>
    );
  }
}

export default Register;
