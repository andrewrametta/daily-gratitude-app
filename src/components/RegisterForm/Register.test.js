import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RegisterForm from "./RegisterForm";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`RegisterForm component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<RegisterForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
