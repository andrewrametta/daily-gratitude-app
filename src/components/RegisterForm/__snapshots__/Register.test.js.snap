// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders the complete page 1`] = `
<div>
  <section>
    <header>
      <h3>
        Register Now
      </h3>
    </header>
    <form
      className="register-form"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="username"
        >
          Username 
        </label>
        <input
          id="username"
          name="username"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="password"
        >
          Password 
        </label>
        <input
          id="password"
          name="password"
          type="password"
        />
      </div>
      <div>
        <label
          htmlFor="confirm-password"
        >
          Confirm Password 
        </label>
        <p>
          Password must be atleast 8 characters and include a number, a capitol letter, and a special character
        </p>
        <input
          id="confirm-password"
          name="confirmPassword"
          type="password"
        />
      </div>
      <button
        type="submit"
      >
        Register
      </button>
    </form>
  </section>
</div>
`;
